src/components/picker/index.tsx

```diff
@@ -298,14 +318,14 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
         /* @ts-expect-error */
         <ExpandableOverlay
           ref={pickerExpandable}
-          useDialog={useDialog || useWheelPicker}
+          useDialog={isDialog || isWheelPicker}
```

Comment:
This should be returned from the hook in the form of type. What I mean is that the hook should be the one source that determines the type of the picker and returns it. Here you should only check if type == dialog.
There are more cases like this in this file